jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/EndPoint.java

```diff
@@ -318,6 +318,30 @@ default void write(Callback callback, SocketAddress address, ByteBuffer... buffe
         write(callback, buffers);
     }
 
+    @Override
+    default void write(boolean last, ByteBuffer byteBuffer, Callback callback)
+    {
+        if (last)
+        {
+            write(Callback.from(() ->
+            {
+                try
+                {
+                    close();
+                    callback.succeeded();
+                }
+                catch (Throwable t)
+                {
+                    callback.failed(t);
+                }
+            }, callback::failed), byteBuffer);
```

Comment:
Shouldn't the 2nd lambda (`callback::failed`) also try to close?

Reply:
I'm not sure we need to close a endpoint that has failed.  Need to think about it...